aten/src/ATen/TensorUtils.h

```diff
@@ -8,6 +8,9 @@
 
 // These functions are NOT in Utils.h, because this file has a dep on Tensor.h
 
+#define TORCH_CHECK_TENSOR(cond, ...) \
+  TORCH_CHECK((cond)._is_all_true().item<bool>(), __VA_ARGS__);
```

Comment:
If we want the semantics to be "check if all elements are true", the name of the macro should reflect it. So it should be something like TORCH_CHECK_TENSOR_ALL ?